src/olympia/discovery/models.py

```diff
@@ -30,6 +30,16 @@ class DiscoveryItem(ModelBase):
         blank=True, help_text='Longer text used to describe an add-on. Should '
                               'not contain any HTML or special tags. Will be '
                               'translated.')
+    position = models.PositiveSmallIntegerField(
+        default=None, blank=True, null=True, db_index=True,
+        help_text='Position in the discovery pane, the lower the number, the '
+                  'higher the item will appear in the page. If left blank or '
+                  'if the value is 0, the item will not appear unless part of '
```

Comment:
I know we do this all over our models (and database)... but if we could settle on a negative being consistently stored as `None`, (or `0`, or `''`) our future selves will thank us.

Reply:
The problem is handling migration in a backwards-compatible way. Django's `default` doesn't set the *database* default, the only thing that does it is allowing `NULL`s, which is done through `null=True`. So we need to support `None` at least. That leaves us in an uncomfortable situation as `0` is a valid `PositiveSmallIntegerField`, so we have to do something about it too.

Reply:
Since we have to do this in two pushes anyway, I could allow `NULL` values initially then set `0` for all and disallow `NULL` in the next push.

Reply:
I don't follow about "backwards-compatible" - you're just adding the field now, so set the default at the same time?

Reply:
If I add a database default other than `NULL` it's SQL that is not supported by Django, so we can't fully test it (and it's not compatible with django migrations either). It doesn't matter if I removed the default in the next push though.